operator/pkg/model/translation/envoy_virtual_host.go

```diff
@@ -224,7 +224,15 @@ func envoyHTTPRoutes(httpRoutes []model.HTTPRoute, hostnames []string, hostNameS
 		if hRoutes[0].RequestRedirect != nil {
 			route.Action = getRouteRedirect(hRoutes[0].RequestRedirect, listenerPort)
 		} else {
-			route.Action = getRouteAction(&r, backends, r.Rewrite, r.RequestMirrors)
+			route.Action = getRouteAction(&r, backends, r.BackendHTTPFilters, r.Rewrite, r.RequestMirrors)
+		}
+		if len(backends) == 1 {
+			for _, fn := range hRoutes[0].BackendHTTPFilters {
+				route.RequestHeadersToAdd = append(route.RequestHeadersToAdd, getHeadersToAdd(fn.RequestHeaderFilter)...)
+				route.RequestHeadersToRemove = append(route.RequestHeadersToRemove, getHeadersToRemove(fn.RequestHeaderFilter)...)
+				route.ResponseHeadersToAdd = append(route.ResponseHeadersToAdd, getHeadersToAdd(fn.ResponseHeaderModifier)...)
+				route.ResponseHeadersToRemove = append(route.ResponseHeadersToRemove, getHeadersToRemove(fn.ResponseHeaderModifier)...)
+			}
```

Comment:
Hmm, I don't quite get why this is here and not in `getRouteAction` (there's the same `if` check around line 344). Could you explain why you put it here?

Reply:
Because when there is only one `backend`, the `BackendFilter` is added to the `route`.

However, in the function of `getRouteAction`, there isn't a `route` object.

When there are more than one `backend`, the `BackendFilter` is added to the `clusterWeight`.

Reply:
Ah, right, I see, thanks for that.

Could you put a comment in to that effect? Thanks!

Reply:
OK.